pynubank/nubank.py

```diff
@@ -45,9 +45,19 @@ def _password_auth(self, cpf: str, password: str):
 
     def _save_auth_data(self, auth_data: dict) -> None:
         self.client.set_header('Authorization', f'Bearer {auth_data["access_token"]}')
-        self.feed_url = auth_data['_links']['events']['href']
-        self.query_url = auth_data['_links']['ghostflame']['href']
-        self.bills_url = auth_data['_links']['bills_summary']['href']
+
+        _links = auth_data['_links']
+        self.query_url = _links['ghostflame']['href']
+
+        if 'magnitude' in _links:
```

Comment:
Criar um teste pra validar esses ifs novos (Um teste seria o suficiente para cobrir todos os elses)

Reply:
Então, eu estava pensando sobre esses testes quando vi o code coverage do PR e me veio uma questão: o método `_save_auth_data` é "privado", logo acessá-lo na construção de um teste seria no mínimo duvidoso rs.
Não tenho muito ideia de como esse teste ficaria, pois testes para as funções públicas que consomem a mesma já existem. Considerei mockar baseado em pelo menos duas resposta do servidor de versões diferentes de clientes (uma usando `magnitude` outra `events`, por exemplo) mas ainda teria que acessar o dito método. 

Se pude dar uma força 😄

Reply:
Inclusive, pensei numa solução mais funcional pra esses `if`s, algo do gênero:
```
    def _save_auth_data(self, auth_data: dict) -> None:
        self.client.set_header('Authorization', f'Bearer {auth_data["access_token"]}')

        links = auth_data['_links']
        self.query_url = links['ghostflame']['href']

        feed_url_keys = ['magnitude', 'events']
        bills_url_keys = ['bills_summary', 'savings_account']

        feed_url_key = next(iter(set(feed_url_keys).intersection(links)))
        bills_url_key = next(iter(set(bills_url_keys).intersection(links)))

        self.feed_url = links[feed_url_key]['href']
        self.bills_url = links[bills_url_key]['href']
```
Daí conforme forem descobertas novas chaves de URLs só ir concatenando nas listas `feed_url_keys` e `bills_url_keys`.
Se tu curtir posso alterar lá no meu fork 😄

Reply:
Sim, ele é privado, mas vc pode testar ele através de um dos métodos públicos que usam ele (`authenticate_with_qr_code`, `authenticate_with_cert` ou `authenticate_with_refresh_token`).

Vc pode evoluir o `MockHttpClient` para ter um novo método que permite registrar/sobreescrever o retorno de uma rota para retornar as chaves `magnitude` e `bills_summary`.

Reply:
Sobre a sua outra solução, acho ela boa, vc pode colocar esse código em uma função para ficar mais claro o que ela faz

Algo assim:

```python
def _find_key(haystack: dict, options: dict) -> str:
  return next(iter(haystack.intersection(dict)))
```

Reply:
Subi as correções nos testes, acho que tá legal, ao menos por enquanto que sabemos que existem 2 "modelos" de URLs.